clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -3202,17 +3202,18 @@ private void gracefulCloseTest(ConsumerCoordinator coordinator, boolean shouldLe
             OffsetCommitRequest commitRequest = (OffsetCommitRequest) body;
             return commitRequest.data().groupId().equals(groupId);
         }, new OffsetCommitResponse(new OffsetCommitResponseData()));
+        if (shouldLeaveGroup)
```

Comment:
Could you explain why we need to check `shouldLeaveGroup` after your change?

Reply:
1.Sure, `consumer.close(Duration.ZERO)` is also a solution, thanks.

2.The `gracefulCloseTest` method is only called by two testCases, `testCloseDynamicAssignment` and `testCloseManualAssignment`.
`testCloseDynamicAssignment` uses the subscribe consumption model,
`testCloseManualAssignment` uses the assign consumption model. The difference here is: when `coordinator.close()` is executed, `testCloseDynamicAssignment` will initiate a LeaveGroupRequest in the underlying `AbstractCoordinator#maybeLeaveGroup(String leaveReason)` method, but `testCloseManualAssignment` will not.
So when I prepared `LeaveGroupResponse` here, I added a judgment on `shouldLeaveGroup`.
The previous `gracefulCloseTest` code logic should prepare redundant `LeaveGroupResponse` for `testCloseManualAssignment`.